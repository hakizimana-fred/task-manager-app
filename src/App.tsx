import Layout from './components/Layout'
import { TasksProvider } from './context/TasksContext'
import { ThemeProvider } from "./context/ThemeContext"

function App() {

  return (
    <ThemeProvider>
    <TasksProvider>
      <Layout />
    </TasksProvider>
    </ThemeProvider>
  )
}

export default App
