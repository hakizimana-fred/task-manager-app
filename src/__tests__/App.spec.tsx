import { render, cleanup } from '@testing-library/react';
import App from '../App';

beforeEach(cleanup);

it('should take a snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot(); 
});
