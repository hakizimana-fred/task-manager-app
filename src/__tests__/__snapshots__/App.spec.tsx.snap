// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <nav
    class="flex flex-col md:flex-row items-center justify-between p-4 bg-white dark:bg-darkBg"
    data-testid="navbar"
  >
    <h1
      class="text-xl font-bold dark:text-gray-300 mb-2 md:mb-0"
    >
      Task Manager
    </h1>
    <div
      class="relative"
    >
      <input
        class="pl-10 pr-4 py-2 rounded border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-800 dark:text-gray-300 focus:outline-none focus:ring-2 focus:ring-blue-500"
        placeholder="Search..."
        type="text"
        value=""
      />
      <svg
        class="lucide lucide-search absolute right-2 top-3 w-5 h-5 text-gray-400 dark:text-gray-300"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <path
          d="m21 21-4.3-4.3"
        />
      </svg>
    </div>
    <div
      class="mt-2 md:mt-0"
    >
      <svg
        class="lucide lucide-moon dark:text-gray-300"
        data-testid="moon-icon"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 3a6 6 0 0 0 9 9 9 9 0 1 1-9-9Z"
        />
      </svg>
    </div>
  </nav>
  <div
    class="flex h-screen bg-lightBg dark:bg-darkBg"
  >
    <aside
      class="w-16 md:w-20 bg-white dark:bg-darkBg h-screen p-4 flex flex-col items-center space-y-8"
      data-testid="sidebar"
    >
      <button
        class="flex flex-col items-center text-gray-700 dark:text-gray-300 hover:text-blue-500 dark:hover:text-blue-400"
      >
        <svg
          class="lucide lucide-list h-6 w-6 mb-1"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 12h.01"
          />
          <path
            d="M3 18h.01"
          />
          <path
            d="M3 6h.01"
          />
          <path
            d="M8 12h13"
          />
          <path
            d="M8 18h13"
          />
          <path
            d="M8 6h13"
          />
        </svg>
        <span
          class="text-xs"
        >
          All Tasks
        </span>
      </button>
      <button
        class="flex flex-col items-center text-gray-700 dark:text-gray-300 hover:text-blue-500 dark:hover:text-blue-400"
      >
        <svg
          class="lucide lucide-circle-check-big h-6 w-6 mb-1"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M21.801 10A10 10 0 1 1 17 3.335"
          />
          <path
            d="m9 11 3 3L22 4"
          />
        </svg>
        <span
          class="text-xs"
        >
          Completed
        </span>
      </button>
      <button
        class="flex flex-col items-center text-gray-700 dark:text-gray-300 hover:text-blue-500 dark:hover:text-blue-400"
      >
        <svg
          class="lucide lucide-clock h-6 w-6 mb-1"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <polyline
            points="12 6 12 12 16 14"
          />
        </svg>
        <span
          class="text-xs"
        >
          Pending
        </span>
      </button>
    </aside>
    <main
      class="flex-1 p-6 bg-myGray dark:bg-darkBg overflow-y-auto"
      data-testid="main"
    >
      <div
        class="flex items-center justify-between mb-4"
      >
        <div
          class="relative inline-flex items-center"
        >
          <select
            class="appearance-none border px-4 py-2 rounded text-gray-700 dark:text-gray-300 border-gray-300 dark:border-gray-600 hover:text-blue-500 dark:hover:text-blue-400 bg-transparent pr-8"
          >
            <option
              value="english"
            >
              English
            </option>
            <option
              value="french"
            >
              French
            </option>
          </select>
          <svg
            class="lucide lucide-chevron-down absolute right-2 pointer-events-none h-4 w-4 text-gray-500 dark:text-gray-400"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </div>
      </div>
      <div
        class="bg-white dark:bg-gray-800 rounded-lg shadow p-6"
      >
        <h2
          class="text-lg font-semibold text-gray-800 dark:text-gray-200"
        >
          Tasks
        </h2>
        <div
          class="flex gap-3 flex-wrap"
        />
      </div>
    </main>
  </div>
</DocumentFragment>
`;
